backend/src/shortener/shortener.controller.ts

```diff
@@ -35,6 +36,9 @@ export class ShortenerController {
     }
 
     let shortUrl: string;
+    if(parsedUrl.host === host) {
+      return { newUrl: parsedUrl.href, isExisting: true };
+    }
```

Comment:
I don't really like this approach of using isExisting variable.
We can just throw a message to the client and show it there.
Also I personally thinks that we shouldn't overload the controller.
Maybe we can try to do a custom decorator with class-validator?

Reply:
sure, will make changes.